### PR TITLE
docs: add jwk and jwe example to readme

### PR DESCRIPTION
**Pull request by laszlo-major-tietoevry**
Hey, thanks for the great work with this library. I took a bit of time to solve a problem I had with it, and while the library did let me do everything I was trying to, it was a bit of a challenge to figure out how exactly....

That's why I expanded the README a bit with the information that would have saved me some time - let me know if you don't find it useful.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,6 +319,18 @@ let jwe = try JWE(compactString: compact)
...
 
+If you want to add additional headers beyond the default to the JWE:
+
+```swift
+var header = DefaultJWEHeaderImpl()
+header.keyID = "Hello-keyId"
+header.keyManagementAlgorithm = .rsaOAEP256
+header.encodingAlgorithm = .a256GCM
+let keyJWK = JWK(keyType: .rsa, algorithm: "A256GCM", keyID: rsaKeyId, e: rsaKeyExponent, n: rsaKeyModulus)
+let jwe = try JWE(payload: wrappedPayload, protectedHeader: header, recipientKey: jwk)
```
Comment (beatt83) [inline README.md:330]:
```suggestion
let rsaKeyId = "Hello-keyId"
var header = DefaultJWEHeaderImpl()
header.keyID = rsaKeyId
header.keyManagementAlgorithm = .rsaOAEP256
header.encodingAlgorithm = .a256GCM
let keyJWK = JWK(keyType: .rsa, algorithm: "A256GCM", keyID: rsaKeyId, e: rsaKeyExponent, n: rsaKeyModulus)
let jwe = try JWE(payload: wrappedPayload, protectedHeader: header, recipientKey: jwk)
```

Comment (beatt83) [reply README.md:330]: Just a small suggestion since the keyID should be the same :)

**Comment by beatt83** [comment]
Hey @laszlo-major-tietoevry . First thanks for the contribution, just left a small change suggestion.
Yes your right and while im using I found some usability enhancements that I could add.
But it could use more examples like this for sure.

**Review by beatt83 — APPROVED** [review]

**Comment by laszlo-major-tietoevry** [comment]
Good suggestion, probably missed it while rewriting my PR for the 5th time :) 

Currently I'm using a mix of this and https://github.com/airsidemobile/JOSESwift - but I'd like to converge on one of them, so I'll probably have some more doc/feature PRs if I run into gaps.

**Comment by beatt83** [comment]
> Good suggestion, probably missed it while rewriting my PR for the 5th time :)
> 
> Currently I'm using a mix of this and https://github.com/airsidemobile/JOSESwift - but I'd like to converge on one of them, so I'll probably have some more doc/feature PRs if I run into gaps.

Awesome! I will then

**Comment by beatt83** [comment]
@laszlo-major-tietoevry just merged. Thanks for the contribution. Any improvements you find please tell me :)

**Pull request merged**